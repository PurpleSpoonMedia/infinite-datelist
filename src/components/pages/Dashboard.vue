<template>
  <div>
    <div>
      <div class="text-center text-sm-right pt-2 pb-4">
        <router-link class="btn btn-success" to="/addnew">Add new user</router-link>
      </div>
      <div class="table-responsive-md">
        <table class="table table-striped">
          <thead>
            <tr>
              <th>Name</th><th>Email</th><th>Phone</th><th>Country</th><th>Services</th><th></th>
            </tr>
          </thead>
          <tbody>
            <app-user-list-item
              v-for="user in users"
              :key="user.id"
              :user="user"
            ></app-user-list-item>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</template>

<script>
import UserListItem from '../UserListItem';
export default {
  computed: {
    users() {
      return this.$store.getters.users;
    }
  },
  created() {
    this.$store.dispatch('setError', false);
  },
  components: {
    AppUserListItem: UserListItem
  }
}
</script>

<style lang="sass" scoped>

</style>