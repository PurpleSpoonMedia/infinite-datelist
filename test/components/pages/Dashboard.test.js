import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VueRouter from 'vue-router'
import Dashboard from '../../../src/components/pages/Dashboard.vue';
import users from '../../fixtures/users';

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(VueRouter);

describe('Dashboard.vue', () => {
  let getters, store;

  beforeEach(() => {
    getters = {
      users: () => users,
    };
    store = new Vuex.Store({
      getters
    });
  })

  test('should render Dashboard correctly', () => {
    const wrapper = shallowMount(Dashboard, {
      store,
      localVue
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
