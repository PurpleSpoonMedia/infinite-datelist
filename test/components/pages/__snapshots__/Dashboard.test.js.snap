// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard.vue should render Dashboard correctly 1`] = `
<div>
  <div>
    <div class="text-center text-sm-right pt-2 pb-4">
      <router-link-stub to="/addnew" tag="a" event="click" class="btn btn-success">Add new user</router-link-stub>
    </div>
    <div class="table-responsive-md">
      <table class="table table-striped">
        <thead>
          <tr>
            <th>Name</th>
            <th>Email</th>
            <th>Phone</th>
            <th>Country</th>
            <th>Services</th>
            <th></th>
          </tr>
        </thead>
        <tbody>
          <appuserlistitem-stub user="[object Object]"></appuserlistitem-stub>
          <appuserlistitem-stub user="[object Object]"></appuserlistitem-stub>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
